haskell/src/CLI.hs

```diff
@@ -84,15 +88,61 @@ getURL url =
       url' = if hasScheme then url else "http://" <> url
    in toText url'
 
+mkSubCommand :: String -> String -> Parser a -> O.Mod CommandFields a
+mkSubCommand name doc parser = command name $ info (parser <**> helper) $ progDesc doc
+
+---------------------------------------------------------------
+-- Janitor cli
+---------------------------------------------------------------
+usageJanitor :: Parser (IO ())
+usageJanitor =
+  subparser
+    ( mkSubCommand "update-idents" "Update author identities" janitorUpdateIdent
+    )
+  where
+    janitorUpdateIdent = io <$> parser
+      where
+        parser = (,,) <$> configOption <*> elasticOption <*> workspaceOption
+        io (configPath, elasticUrl, workspaceNameM) = do
+          setEnv "CRAWLERS_API_KEY" "nokey" -- set a fake env var to resolv the configuration
+          config <- Config.loadConfig configPath
+          env <- mkEnv $ getURL elasticUrl
+          case workspaceNameM of
+            Just workspaceName -> do
+              void $ case Config.lookupTenant (Config.getWorkspaces config) workspaceName of
+                Nothing -> print $ "Unable to find the workspace " <> workspaceName <> " in the Monocle config"
+                Just workspace -> runOnWorkspace env workspace
+            Nothing -> traverse_ (runOnWorkspace env) $ Config.getWorkspaces config
+        workspaceOption = optional $ strOption (long "workspace" <> O.help "Workspace name" <> metavar "WORKSPACE")
+        configOption = strOption (long "config" <> O.help "Path to configuration file" <> metavar "CONFIG")
+        elasticOption = strOption (long "elastic" <> O.help "The Elastic endpoint url" <> metavar "ELASTIC_URL")
+        runOnWorkspace env workspace = do
```

Comment:
runOnWorkspace and the body of the io action could be moved to the Janitor module too to avoid doing too much in the cli usage parser.